app/gui/src/model/suggestion_database.rs

```diff
@@ -268,6 +365,51 @@ impl From<language_server::response::GetSuggestionDatabase> for SuggestionDataba
 
 
 
+// ===============
+// === Helpers ===
+// ===============
+
+
+// === QualifiedNameToIdMap helpers ===
+
+/// Swaps the value at `path` in `node` with `value` (sets `value` to `None` if there was no node
+/// at `path`). Then visits nodes on the `path` in reverse order and removes every visited empty
+/// leaf node from its parent.
+///
+/// In this function, a node is defined as empty when it contains a `None` value. A node is a leaf
+/// when its [`is_leaf`] method returns `true`.
+///
+/// The function is optimized to not create new empty nodes if they would be deleted by the
+/// function before it returns.
+fn swap_value_and_traverse_back_pruning_empty_subtrees<P, I>(
+    node: &mut HashMapTree<entry::QualifiedNameSegment, Option<entry::Id>>,
+    mut path: P,
+    value: &mut Option<entry::Id>,
+) where
```

Comment:
If this is a helper, I'd like to see information why it is needed. The docs repeat information from replace_value_and_traverse_back_pruning_empty_subtrees, so I had an impression that it could be just inlined there.